go/ai/tools.go

```diff
@@ -47,26 +47,25 @@ func (t ToolName) Name() string {
 	return (string)(t)
 }
 
-// tool is the internal implementation of the Tool interface.
-// It holds the underlying core action and allows looking up tools
-// by name without knowing their specific input/output types.
+// tool is an action with functions specific to tools.
+// It embeds [core.Action] instead of [core.ActionDef] like other primitives
+// because the inputs/outputs can vary and the tool is only meant to be called
+// with JSON input anyway.
 type tool struct {
 	core.Action
```

Comment:
core.ActionDef[*ToolInput, *ToolOutput, struct{}]?

Reply:
`tool` is the exception since it does not have predefined inputs and outputs like the other `ai` primitives so we can't get hard types when reading back from the registry.